container/extender/src/main/java/org/opennms/karaf/extender/KarDependencyHandler.java

```diff
@@ -0,0 +1,166 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.karaf.extender;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.karaf.features.FeaturesService;
+import org.apache.karaf.features.Repository;
+import org.apache.karaf.kar.KarService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KarDependencyHandler implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(KarDependencyHandler.class);
+
+    private static final int KAR_LIST_SLEEP_MS = 5000;
+
+    private final List<Feature> features;
+    private final KarService karService;
+    private final FeaturesService featuresService;
+
+    private static final String FEATURE_CONFIG_FILE = "features.cfg";
+    private static final String KAR_STORAGE = System.getProperty("karaf.data") + File.separator + "kar";
```

Comment:
Is this where it ends up getting copied to _after_ `/deploy/` through some secret Karaf voodoo?  Or should this be `deploy` rather than the `karaf.data` directory?

Reply:
Yeah. The `kar` service in Karaf will extract/install the `.kar` files here.